components/Msig/ApproveCancel.tsx

```diff
@@ -0,0 +1,147 @@
+import PropTypes from 'prop-types'
+import { useState, useMemo, Context, useEffect } from 'react'
+import { useRouter } from 'next/router'
+import { Message } from '@glif/filecoin-message'
+import { FilecoinNumber } from '@glif/filecoin-number'
+import {
+  navigate,
+  convertAddrToPrefix,
+  decodeActorCID,
+  useActorQuery,
+  useWallet,
+  Transaction,
+  Parameters,
+  MsigMethod,
+  TxState,
+  WalletProviderOpts,
+  PendingMsgContextType,
+  MsigTransaction
+} from '@glif/react-components'
+
+import { useMsig } from '../../MsigProvider'
+import { useWasm } from '../../lib/WasmLoader'
+import { PAGE } from '../../constants'
+import { logger } from '../../logger'
+
+type MsigTxWithApprovals = MsigTransaction & { approvalsUntilExecution: number }
+
+export const ApproveCancel = ({
+  method,
+  walletProviderOpts,
+  pendingMsgContext
+}: ApproveCancelProps) => {
+  const router = useRouter()
+  const wallet = useWallet()
+  // @ts-expect-error
+  const { serializeParams } = useWasm()
+  const { Address, AvailableBalance } = useMsig()
+
+  // Transaction states
+  const [txState, setTxState] = useState<TxState>(TxState.LoadingMessage)
+  const [txFee, setTxFee] = useState<FilecoinNumber | null>(null)
+
+  // Get transaction info from url
+  const { proposal } = router.query
+  const transaction = useMemo<MsigTxWithApprovals | null>(() => {
+    try {
+      return JSON.parse(decodeURI(proposal as string))
+    } catch (e) {
+      setTxState(TxState.LoadingFailed)
+      return null
+    }
+  }, [proposal])
+
+  // Create message from input
+  const message = useMemo<Message | null>(() => {
+    try {
+      return transaction
+        ? new Message({
+            to: Address,
+            from: wallet.address,
+            nonce: 0,
+            value: 0,
+            method,
+            params: Buffer.from(
+              serializeParams({
+                TxnID: transaction.id,
+                ProposalHashData: transaction.proposalHash
+              }),
+              'hex'
+            ).toString('base64'),
+            gasPremium: 0,
+            gasFeeCap: 0,
+            gasLimit: 0
+          })
+        : null
+    } catch (e) {
+      logger.error(e)
+      return null
+    }
+  }, [Address, wallet.address, method, transaction, serializeParams])
+
+  // Get actor data from transaction
+  const { data: actorData, error: actorError } = useActorQuery({
+    variables: { address: convertAddrToPrefix(transaction?.to.robust) },
+    skip: !transaction?.to.robust
+  })
```

Comment:
I think here we should fallback to the ID address if robust isnt available

I _think_ if the user enters an ID address, thats what `Address` will be

Reply:
Fixed in https://github.com/glifio/safe/pull/107